tscached/shadow.py

```diff
@@ -0,0 +1,33 @@
+import logging
+
+
+def should_add_to_readahead(config, referrer, headers):
+    """ Should we add this KQuery for readahead behavior?
+        config: dict representing the top-level tscached config
+        referrer: str, from the http request
+        headers: dict, all headers from the http request
+        returns: boolean
+    """
+    if headers.get(config['shadow']['http_header_name'], None):
+        return True
+
+    for substr in config['shadow']['referrer_blacklist']:
+        if substr in referrer:
+            return False
+    return True
```

Comment:
So every request that doesn't contain a blacklist substr is going to be added for read ahead for the given TTL (I assume the TTL is how long it's maintained in the read-ahead list)?